### PR TITLE
docs: autoincrement apiLevel in example code

### LINKED ISSUE #5758: feat: auto-increment apiLevel in docs
# Overview
For better or for worse, our users are not primarily software engineers and thus take [less than optimal approaches](https://en.wikipedia.org/wiki/Copy-and-paste_programming) to writing their own python protocols.

Reading the docs for the first time (wanting to spend more time doing science than learning syntax), a novice programmer might be tempted to copy and paste the very first (or any) block of code they see that fits their purpose.

Even though what the user has done is bad practice, we should still try and help make their lives easier by automatically incrementing the `apiLevel` in all example code in the docs to the most current version to ensure they get the best starting experience.

The current process is not automated (eg: https://github.com/Opentrons/opentrons/pull/5054 https://github.com/Opentrons/opentrons/pull/5507 ) and thus the current `apiLevel` is outdated in example code everywhere in the docs ... [even on the page containing the description](https://docs.opentrons.com/v2/versioning.html) of the current `apiLevel`!

To give our customers the best experience it's important new users start their code on the most recent `apiLevel` as it generally contains important fixes. (e.g. https://github.com/Opentrons/opentrons/issues/5448 and particularly https://github.com/Opentrons/opentrons/pull/5039 !!) even if they are using less than optimal coding practices.

# Implementation details
It is possible to specify a `rst_prolog` string in the [Sphinx configuration](https://github.com/Opentrons/opentrons/blob/edge/api/docs/v2/conf.py) and thus automatically supply the latest `apiLevel` in the build for replacement using the `|substitution|` reStructuredText syntax.

This is the currently used method to supply the `|version|` and `|release|` variables to the docs, and we should implement `|apiLevel|` as well.

We could either pull in the `MAX_SUPPORTED_VERSION` to `package.json` or just `str(opentrons.protocol_api.MAX_SUPPORTED_VERSION)` directly in the Sphinx `conf.py` to substitute the most current `apiLevel` in the example code used in the docs.

┆Issue is synchronized with this [Wrike Task](https://www.wrike.com/open.htm?id=535451160) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by nickcrider**
# Overview

Provides a mechanism that automatically updates the `apiLevel` in example code in the docs to the `MAX_SUPPORTED_VERSION` 

Closes #5758

# Changelog

- adds [an extension to Sphinx](https://github.com/adamtheturtle/sphinx-substitution-extensions) that allows for a `:substitutions:` option to be added to `code-block` directives
- automatically gets the `MAX_SUPPORTED_VERSION` and creates a `|apiLevel|` rst substitution variable containing the string
- updates all existing code blocks to use the latest API Level

# Review requests

- make sure the `Pipfile` stuff is correct
- This creates some additional Sphinx errors, does that interfere with the work going on in #6135

# Risk assessment

Low, only affects docs

When adding new example code to the docs, authors will have to remember to:
- add the `:substitutions:` option to the `code-block` directive
- as well as `metadata = {'apiLevel': '|apiLevel|'}` or `get_protocol_api('|apiLevel|')`

### CODE REVIEW

**Comment by nickcrider** [comment]
Thanks for the guidance @sfoster1, and yeah I got a little aggressive with the find-replace on the first pass 😀 

Take a look now and let me know what you think!

File: api/Pipfile [hunk 1]
```diff
@@ -26,6 +26,8 @@ wheel = "==0.30.0"
 "c445fee" = {editable = true, path = "./../shared-data/python"}
 pytest-aiohttp = "*"
 typeguard = "*"
+sphinx-substitution-extensions = "*"
+v = {version = "*", editable = true}
```
Comment (sfoster1) [inline api/Pipfile:30]: What's this? Maybe a typo in a `pipenv install` command?

Comment (sfoster1) [reply api/Pipfile:30]: this is that package: https://pypi.org/project/v/

Try removing it with `pipenv uninstall`

File: api/docs/v2/index.rst [hunk 2]
```diff
@@ -54,6 +54,7 @@ The design goal of this API is to make code readable and easy to understand. For
 If we were to rewrite this with the Python Protocol API, it would look like the following:
 
 .. code-block:: python
+    :substitutions:
```
Comment (sfoster1) [inline api/docs/v2/index.rst:57]: Ah, I like this a lot better than the `code-block-substitutions` directive

File: api/Pipfile (hunk 1 above)
Comment (nickcrider) [reply api/Pipfile]: Looked at my command history I think `pipenv install -dev Sphinx-Substitution-Extensions` probably caused that. Thanks for catching.

File: api/docs/v2/writing.rst [hunk 3]
```diff
@@ -83,7 +83,7 @@ Jupyter Installation
 
 You must make sure that you install the ``opentrons`` package for whichever kernel and virtual environment the notebook is using. A generally good way to do this is
 
-.. code-block:: shell
+.. prompt:: bash
```
Comment (sfoster1) [inline api/docs/v2/writing.rst:86]: This is actually a jupyter ipython notebook command rather than bash. maybe `.. prompt:: python` would be better?

**Comment by sfoster1** [comment]
> Looking at the build failure reason, I think the latex build stopped; also, based on some sphinx error output, I _think_ it's not properly loading the substitution extensions. I'll take a look at why; I think we may need to mess around with the sys path for the sphinx runs a bit more.

No, the (incredibly annoying) problem is with the backslash in the windows path in the example of running `opentrons-simulate` with custom labware.

File: api/docs/v2/writing.rst [hunk 4]
```diff
@@ -132,7 +132,7 @@ directory in which you run it. You can change the directory
 ``opentrons_simulate`` searches for custom labware with the
 ``--custom-labware-path`` option:
 
-.. code-block:: shell
+.. prompt:: bash
```
Comment (sfoster1) [inline api/docs/v2/writing.rst:135]: As annoying as it is, I'd recommend changing this back to a `.. code-block:: shell` to make the pdf rendering work again

File: api/docs/v2/writing.rst (hunk 3 above)
Comment (nickcrider) [reply api/docs/v2/writing.rst:86]: Got this to work!

**Comment by nickcrider** [comment]
Made the requested changes and also as a bonus enabled the `:substitution-code:` directive for the applicable line. Should be good to go!

**Pull request merged**